### PR TITLE
Update dependencies & Rust 2018

### PR DESCRIPTION
**Pull request by ccakes**
This PR does the following:

- Update all dependencies, including `parquet` & friends, to current versions
- Minor change to the metadata getters to deal with changes in the `parquet` crate
- Update the Rust edition to 2018
- Bump version to 1.0.0 to keep inline with parquet (feel free to not do this if you would prefer to stay 0.x)
- Add Github Actions workflow to build & release

### CODE REVIEW

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-nightly-2019-06-14
+nightly
```
Comment (FabioBatSilva) [inline rust-toolchain]: I had this fixed BC the latest nightly usually doesn't have all the components we are using..
https://rust-lang.github.io/rustup-components-history

I think `clippy` /  `fmt` is going to fail on nightly eventually if not currently failing

File: src/reader.rs [hunk 2]
```diff
@@ -155,11 +155,11 @@ impl ParquetFile {
             .unwrap_or_else(|| Err(Error::from(self.path.to_path_buf())))
     }
 
-    pub fn metadata(&self) -> Result<ParquetMetaDataPtr> {
+    pub fn schema(&self) -> Result<Type> {
```
Comment (FabioBatSilva) [inline src/reader.rs:158]: I believe there are other places where this func is being used.

if you run the following command you should see where it fails: 
```
cargo clean
cargo test
```

File: Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 [package]
 name = "xpq"
-version = "0.1.2"
+version = "1.0.0"
```
Comment (FabioBatSilva) [inline Cargo.toml]: Let's make it `0.2.0`.. i Don't think we can call it stable yet

**Review by FabioBatSilva — CHANGES_REQUESTED** [review]
Thanks for the PR @ccakes

i've been procrastinating on doing this for a while.

File: src/reader.rs (hunk 2 above)
Comment (ccakes) [reply src/reader.rs:158]: The only other places this is used is in the tests - if it's not used in the application code anymore are you happy for me to just fix up the test?

The reason for the change is that it's no longer possible (that I can find..) to acquire an owned [`parquet::file::metadata::ParquetMetaData`](https://docs.rs/parquet/1.0.0/parquet/file/metadata/struct.ParquetMetaData.html), only a reference.

File: rust-toolchain (hunk 1 above)
Comment (ccakes) [reply rust-toolchain]: Ok, I've pinned it at the latest version with rustfmt which at the moment is 2020-07-27

**Comment by ccakes** [comment]
I also pulled out the Github Actions stuff and updated the tests based on my comment above but let me know if you want to handle this differently.

I also removed `format_doc_comments` from rustfmt.toml as it doesn't seem to be supported by rustfmt anymore? 🤷‍♂️ 
```
$ cargo fmt -- --check
Warning: Unknown configuration option `format_doc_comments`
```

**Comment by FabioBatSilva** [comment]
Thanks @ccakes 👍

**Pull request merged**